### PR TITLE
fix(j-s): Defender Link

### PR DESCRIPTION
**Pull request by gudjong**
# Defender Link

[Fela "afrita hlekk fyrir verjendur" hjá sækjanda þar til fyrirtöku tími hefur verið boðaður](https://app.asana.com/0/1199153462262248/1203844632938806/f)

## What

- Disables the button "Afrita hlekk fyrir verjanda" until the case has been scheduled by the court.

## Why

- The defender will not have access until the case has been scheduled.

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by unakb — APPROVED** [review]
Just one question but other than that lgtm

edit: Just noticed there are failing tests

**Review by unakb — CHANGES_REQUESTED** [review]
Some e2e tests need to be fixed because they're trying to click the now disabled button

File: apps/judicial-system/web/src/routes/Prosecutor/InvestigationCase/Overview/Overview.tsx [hunk 1]
```diff
@@ -355,6 +355,10 @@ export const Overview: React.FC = () => {
               <CopyLinkForDefenderButton
                 caseId={workingCase.id}
                 type={workingCase.type}
+                disabled={
+                  workingCase.state !== CaseState.RECEIVED ||
+                  !workingCase.courtDate
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Prosecutor/InvestigationCase/Overview/Overview.tsx:360]: It should not happen, but it has happened. When the case gets a court case number, then the case should transition to a received state. That happens long before a court date is set, so checking only for the court date should in theory be fine. However, the transition has failed in the past and I am more confident double checking here. At some point, we should also add checks in the client that prevent navigating to the court date page if the case is not in the received state.

[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/web/src/routes/Prosecutor/InvestigationCase/Overview/Overview.tsx:360]: Ok makes sense, and good to know 😊

[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Prosecutor/InvestigationCase/Overview/Overview.tsx:360]: Also, the logic that gives a defender access on the server side checks that the case is received, so matching that logic here is right.

**Review by unakb — APPROVED** [review]

**Pull request merged**